hal/armv7m/imxrt/_init.S

```diff
@@ -179,10 +179,20 @@ _exceptions_dispatch:
 	cpsid if
 
 	mrs r0, psp
+	mov r2, sp
+	mov r3, r0
 	stmdb sp!, {r0, r4-r11, lr}
 
 	mrs r0, ipsr
 	mov r1, sp
+
+	cmp lr, RET_THREAD_PSP
+	bne 1f
+	cmp r3, NULL
+	beq 1f
+	mov r2, r3
```

Comment:
Assuming `NULL == 0` you can do it like this
```suggestion
	cbz r3, 1f
	cmp lr, RET_THREAD_PSP
	it eq
	moveq r2, r3
```

Reply:
Fixed, thanks!